### PR TITLE
Election Statistics Report

### PR DESCRIPTION
**Pull request by JohnMwashuma**
### Overview
---
Election statistics report per election and per ballot.

### Context
---
Implements the report in slide one of this google sheet file https://docs.google.com/spreadsheets/d/1YnykTc6-MYjokM15VRmZcSWnBS-JESA1eTEhDEivVu8/edit#gid=0

### Task check list
---
- [x] Generate overview election statics per election level.
- [x] Generate ballots progress statistics per election level.
- [x] Add Overall as the last row which sums column values.
- [x] Remove `CHOICES` option from the gender filter.
- [x] Optimize queryset ballots progress statistics per election level.
- [x] Add tests.

### CODE REVIEW

**Review by JohnMwashuma — COMMENTED** [review]
@lincmba So the queryset is still the same, it needs to be optimized for performance. Kindly have a look.

File: tally_ho/apps/tally/views/reports/election_statistics_report.py [hunk 1]
```diff
@@ -0,0 +1,380 @@
+import json
+from django.http import JsonResponse
+from django.views.generic import TemplateView
+from django.urls import reverse
+from django_datatables_view.base_datatable_view import BaseDatatableView
+from django.db.models import Count, Sum, F, Case, When, IntegerField, Value
```
[the remark this answers is not in the document]
Comment (JohnMwashuma) [reply tally_ho/apps/tally/views/reports/election_statistics_report.py]: fixed

[the remark this answers is not in the document]
Comment (JohnMwashuma) [reply tally_ho/apps/tally/views/reports/election_statistics_report.py]: fixed

[the remark this answers is not in the document]
Comment (JohnMwashuma) [reply tally_ho/apps/tally/views/reports/election_statistics_report.py]: fixed

File: tally_ho/apps/tally/views/reports/election_statistics_report.py [hunk 2]
```diff
@@ -0,0 +1,347 @@
...
+            ballot_election_statistics[
+                'percentage_turnout_in_stations_counted'] =\
+                round(
+                    100 * stations_counted / election_statistics[
+                        'stations_expected'],
+                    2)
+            ballot_election_statistics[
+                'percentage_turnout_in_stations_counted'] =\
+                    round(
+                    100 * stations_counted / election_statistics[
```
[the remark this answers is not in the document]
Comment (JohnMwashuma) [reply tally_ho/apps/tally/views/reports/election_statistics_report.py]: fixed

File: tally_ho/apps/tally/views/reports/election_statistics_report.py [hunk 3]
```diff
@@ -0,0 +1,347 @@
...
+        # Calculate turnout percentage
+        if stations_counted != 0:
+            ballot_election_statistics['voters_in_counted_stations'] =\
+                voters_in_counted_stations
+            ballot_election_statistics['registrants_in_stations_counted'] =\
+                registrants_in_stations_counted
+            ballot_election_statistics[
+                'percentage_turnout_in_stations_counted'] =\
+                round(
+                    100 * stations_counted / election_statistics[
```
[the remark this answers is not in the document]
Comment (JohnMwashuma) [reply tally_ho/apps/tally/views/reports/election_statistics_report.py]: fixed

File: tally_ho/apps/tally/tests/views/reports/test_election_statistics_report.py [hunk 4]
```diff
@@ -0,0 +1,122 @@
+import json
```
[the remark this answers is not in the document]
Comment (JohnMwashuma) [reply tally_ho/apps/tally/tests/views/reports/test_election_statistics_report.py]: fixed

**Comment by p-netm** [comment]
Looks good, Just a few issues.

The %turnout seems to be off, it seems it calculates it as `registrants/voters` while I think it should be `voters/registrants.`:
Screenshot from 2023-08-16 12-49-11h

Screenshot from 2023-08-16 12-48-46

======

Also, in the below screenshot:
image

The total voters are `>0`, but `male voters + female voters = 0`, this is objectively wierd, I understand there is the `Unisex` votes, which if included would make the displayed data be more complete, I however think if I remember correctly the consesus was to split the Unisex between male and female!!!!,

**Review by p-netm — APPROVED** [review]

**Pull request merged**